add grpc client reconnect every 5 minutes

danielamiao opened the pull request.

frenchfrywpepper commented on common/src/main/java/com/lightstep/tracer/shared/CollectorClient.java:
```diff
@@ -31,12 +32,13 @@ public CollectorClient(AbstractTracer tracer, String host, int port) {
    */
   public CollectorClient(AbstractTracer tracer, ManagedChannelBuilder<?> channelBuilder) {
     this.tracer = tracer;
+    this.channelBuilder = channelBuilder;
```
I have a slight preference for de-deduplicating these two lines that initialize channel and blockingStub, but I won't insist.

jmacd commented on common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java:
```diff
@@ -29,6 +29,7 @@
     // Maximum interval between reports
     private static final long DEFAULT_CLOCK_STATE_INTERVAL_MILLIS = 500;
     private static final int DEFAULT_REPORT_TIMEOUT_MILLIS = 10 * 1000;
+    private static final int DEFAULT_CLIENT_RESET_INTERVAL_MILLIS = 5 * 60 * 1000; // 5 min
```
I haven't thought about whether Android libraries want this feature, or not. (Maybe it will hurt battery performance?)

danielamiao replied:
this is a good point, plus android clients probably don't need this at all because they can't realistically generate a high load. I'll add in an additional config which turns this off by default in the Android tracer.

danielamiao commented:
Would appreciate another look - I addressed PR comments by adding a config to turn resetClient on/off, it's off by default in the Android tracer impl, thanks!
